### PR TITLE
Feature/issue 18

### PR DESCRIPTION
**Pull request by raynasr**
Listing Screen isn't accessible from Navigation yet (we will implement it once we have actual listings to browse to.) Fake listing present to test in NavController. Set destination to "Listing".
Added tests for every screen.
Home now has 4 buttons: Add Listing (different from Listing), Cart, About, Browse.
Developed NavigationTest further

### CODE REVIEW

**Review by tepoorte — APPROVED** [review]
It looks good to me, still you can add some private methods in order to make the code more understandable for each screen

File: app/src/androidTest/java/com/studhub/app/ui/ListingScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+    @Test
+    fun testListingScreen() {
+        val mockOnContactSellerClick: () -> Unit = {}
+        val mockOnFavouriteClick: () -> Unit = {}
+
+        val listing = Listing(
+            id = 1,
```
Comment (edouardmichelin) [inline app/src/androidTest/java/com/studhub/app/ui/ListingScreenTest.kt:27]: The `id` field is not a `String`

File: app/src/main/java/com/studhub/app/ui/ListingScreen.kt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                Spacer(modifier = Modifier.height(35.dp))
+                Text(
+                    text = "Price:",
+                    style = typography.titleMedium,
+                    color = contentColorFor(MaterialTheme.colorScheme.surface)
+                )
+                Spacer(modifier = Modifier.height(8.dp))
+                Text(
+                    // $ + price
+                    text = "$${listing.price}",
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/ui/ListingScreen.kt:86]: Prices will be in CHF ;)

File: app/src/main/java/com/studhub/app/ui/ListingScreen.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.studhub.app.domain.model.Listing
+import com.studhub.app.ui.theme.StudHubTheme
+
+@Composable
+fun ListingScreen(listing: Listing,  onContactSellerClick: () -> Unit,
+                  onFavouriteClick: () -> Unit) {
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/ui/ListingScreen.kt:17]: Weird alignment, I think the following would look nicer:
```kt
fun listingScreen(
    listing: Listing,
    onContactSellerClick: () -> Unit,
    onFavouriteClick: () -> Unit
) {
    // code
}
```

Comment (edouardmichelin) [reply app/src/main/java/com/studhub/app/ui/ListingScreen.kt:17]: image

cf this from [https://kotlinlang.org/docs/coding-conventions.html#functions](https://kotlinlang.org/docs/coding-conventions.html#functions)

File: app/src/androidTest/java/com/studhub/app/ui/ListingScreenTest.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                listing = listing,
+                onContactSellerClick = mockOnContactSellerClick,
+                onFavouriteClick = mockOnFavouriteClick
+            )
+        }
+        // wait for UI
+        composeTestRule.waitForIdle()
+
+        composeTestRule.onNodeWithText(listing.name).assertIsDisplayed()
+        composeTestRule.onNodeWithText("$${listing.price}").assertIsDisplayed()
```
Comment (edouardmichelin) [inline app/src/androidTest/java/com/studhub/app/ui/ListingScreenTest.kt:49]: Don't forget to also modify the dollar sign from the test if you modify it in the other class

**Review by edouardmichelin — CHANGES_REQUESTED** [review]
Added some comments. Otherwise LGTM

**Comment by raynasr** [comment]
Changes done

File: app/src/main/java/com/studhub/app/ui/ListingScreen.kt [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.studhub.app.domain.model.Listing
+import com.studhub.app.ui.theme.StudHubTheme
+
+@Composable
+fun ListingScreen(listing: Listing,
+                  onContactSellerClick: () -> Unit,
+                  onFavouriteClick: () -> Unit
+)
```
Comment (edouardmichelin) [inline app/src/main/java/com/studhub/app/ui/ListingScreen.kt:19]: `CTRL + ALT + L` ;)

Comment (edouardmichelin) [reply app/src/main/java/com/studhub/app/ui/ListingScreen.kt:19]: The offset seems a bit large

**Review by edouardmichelin — CHANGES_REQUESTED** [review]
Just a minor change, otherwise LGTM

**Review by edouardmichelin — APPROVED** [review]
All good

**Pull request merged**